### PR TITLE
doc: update Python req. for sphinx

### PR DESCRIPTION
**Pull request by r-rojo**
* Upgrade the sphinx/requirements.txt to use latest versions of the
  following modules:
  * requests==2.24.0
  * sphinx==3.2.1
  * sphinx-markdown==1.0.2
  * recommonmark==0.6.0
  * breathe==4.23.0
* Update the sphinx configuration to use recommonmark instead
  of pandoc_markdown.
* Tweak some of the .rst files according to some warnings emitted by
  sphinx.
  * TODO: Fix other warnings in another pull request.
* Tweak build-documentation.sh script to simply build the documentation
  but not upload it to github.io
* Add github workflow job for building documentation
  * This job uses opae/doc-builder container to run build-documentation.sh
  * This job can call push-documentation.sh if on master or a tag
  * Add a link checker step to look for broken links in docs
  * Archive html docs and link checker result file

### CODE REVIEW

File: scripts/push-documentation.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
```
Comment (tswhison) [inline scripts/push-documentation.sh:2]: This should be a bash.  Vanilla sh doesn't support function definitions.

**Review by tswhison — APPROVED** [review]

**Review by asgardkm — DISMISSED** [review]
nice!

File: docker/docs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:latest
+RUN apt-get update
+RUN apt-get install apt-utils
+RUN apt-get install -y tzdata && dpkg-reconfigure --frontend noninteractive tzdata
+RUN apt-get install -y build-essential cmake git libjson-c-dev uuid-dev git python3-dev python3-wheel python3-pip doxygen
+WORKDIR /root
+ENTRYPOINT [ "/bin/bash", "-c" ]
```
Comment (tswhison) [inline docker/docs/Dockerfile:7]: Doesn't -c require a command to execute, or is it different in bash.  I know that sh -c requires a command.

Comment (r-rojo) [reply docker/docs/Dockerfile:7]: It requires a command, the point of this container's entry point is basically to run a bash command in the context of the container.

**Review by asgardkm — APPROVED** [review]

**Review by tswhison — APPROVED** [review]

**Pull request merged**